polkadot/runtime/parachains/src/mock.rs

```diff
@@ -388,6 +385,101 @@ impl ValidatorSetWithIdentification<AccountId> for MockValidatorSet {
 	type IdentificationOf = FoolIdentificationOf;
 }
 
+pub mod mock_assigner {
+	pub use pallet::*;
+	use super::*;
+	#[frame_support::pallet]
+	pub mod pallet {
+		use super::*;
+
+		#[pallet::pallet]
+		#[pallet::without_storage_info]
+		pub struct Pallet<T>(_);
+
+		#[pallet::config]
+		pub trait Config: frame_system::Config + configuration::Config + paras::Config {}
+
+		/// Creates an empty on demand queue if one isn't present in storage already.
+		#[pallet::type_value]
+		pub(super) fn OnDemandQueueOnEmpty<T: Config>() -> VecDeque<V0Assignment> {
+			VecDeque::new()
+		}
+
+		#[pallet::storage]
+		pub(super) type MockAssignerQueue<T: Config> =
+			StorageValue<_, VecDeque<V0Assignment>, ValueQuery, OnDemandQueueOnEmpty<T>>;
+	}
+
+	impl<T: Config> Pallet<T>{
+		pub fn add_on_demand_order(
```

Comment:
Why is our mock designer an on-demand assigner?

Reply:
Very good point. This function isn't actually the problem on that front though. It was implementing the `AssignmentProvider` api as if `MockAssigner` was just an `OnDemandAssigner`.

I added the legacy auction assigner functionality to it, making it more of a two in one assigner that still supports the testing specific on demand calls, like `add_on_demand_order`.